File: src/exercise/python_solution/find_max_and_even_odd.py
max2 = None
list2 = None
x = None
num = None
max2 = 0
list2 = [5, 2, 8]
for num in list2:
  if num % 2 == 0:
    print('Number is even')
  else:
    print('Number is odd.')
  if num >= max2:
    max2 = num
    print('This is the current max.')

File: src/exercise/python_unittest/find_max_and_even_odd.js
/* eslint-disable camelcase */
const find_max_and_even_odd_unittest = `
import unittest
from io import StringIO
import sys

def method():
{parsons_puzzle}
class myTests(unittest.TestCase):
    @patch('builtins.input',side_effect=['4','9','2','-1'])
    def testOne(self, input):
        capturedOutput = StringIO()          # Create StringIO object
        sys.stdout = capturedOutput                   #  and redirect stdout.
        method()                                  # Call unchanged function.
        sys.stdout = sys.__stdout__                   # Reset redirect.
        outputArray = capturedOutput.getvalue().split('\n')
        if (len(outputArray)>1):
            outputArray.pop()
        self.assertEqual(outputArray,
           ['Number is even.','This is the current max.','Number is odd.','This is the current max.','Number is even.'],)
        return 'success'


_test_result = myTests().testOne()`;

export default find_max_and_even_odd_unittest;
